EIPS/eip-7504.md

```diff
@@ -0,0 +1,421 @@
+---
+eip: 7504
+title: Dynamic Contracts
+description: Client-friendly one-to-many proxy contracts.
+author: Krishang Nadgauda (@nkrishang)
+discussions-to: https://ethereum-magicians.org/t/erc-7504-dynamic-contracts/15551
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-08-25
+---
+
+## Abstract
+
+Building upgradeable contracts in the ‘proxy pattern’ — a proxy contract forwarding calls to an implementation contract — is now industry standard.
+
+![diagram showing a proxy contract that forwards calls to an implementation contract.](../assets/eip-7504/proxy-with-single-impl.png)
+
+Standards like [ERC-2535](./eip-2535.md) Diamonds introduce the concept of a proxy contract using many implementation smart contracts, where the proxy selects one out of many implementation contracts to delegateCall.
+
+This proposal standardizes how ‘proxy-with-many-implementations’ or ‘one-to-many’ proxy contracts can be written with client friendliness and adoption in mind. Concretely, this proposal standardizes how one-to-many proxy should expose their true ABI for straightforward contract interaction via clients.
+
+Going forward, we refer to such one-to-many proxy contracts as dynamic contracts.
+
+## Motivation
+
+This proposal does not introduce any novel technique of writing smart contracts. Dynamic contracts, or ‘one-to-many’ proxy contracts have been introduced before in prior EIP standards, or in custom implementations.
+
+This proposal aims to propel the adoption of dynamic contracts by making them easy to grasp, and straightforward to interact with via clients.
```

Comment:
You've used clients quite a few times by this point, but I'm still not sure what you mean. Do you mean "ethereum clients" like geth, or wallets, or simply "anything that calls the proxy"?

Reply:
I've now clarified in key places that: "'Clients' refers to scripts, websites and any such domain for code that calls a smart contract."